lib/hammer_cli_katello/repository.rb

```diff
@@ -201,7 +201,35 @@ class CreateCommand < HammerCLIKatello::CreateCommand
              :attribute_name => :option_unprotected,
              :format => HammerCLI::Options::Normalizers::Bool.new
 
+      option "--ssl-client-cert", "SSL_CLIENT_CERT",
+             _("Name of content credential containing client certificate"),
+             :attribute_name => :option_ssl_client_cert
+
+      option "--ssl-client-key", "SSL_CLIENT_KEY",
+             _("Name of content credential containing client key"),
+             :attribute_name => :option_ssl_client_key
+
+      option "--ssl-ca-cert", "SSL_CA_CERT",
+             _("Name of content credential containing CA certificate"),
+             :attribute_name => :option_ssl_ca_cert
+
       build_options :without => [:unprotected]
+
+      def request_params
+        params = super
+
+        params["ssl_client_cert_id"] = get_gpg_key_id("ssl_client_cert") if option_ssl_client_cert
+        params["ssl_client_key_id"] = get_gpg_key_id("ssl_client_key") if option_ssl_client_key
+        params["ssl_ca_cert_id"] = get_gpg_key_id("ssl_ca_cert") if option_ssl_ca_cert
```

Comment:
Maybe these should be memoized in case a user passes both ID and name? Usually we'd have an option_validation to ensure users cannot pass both ID and name.

Reply:
That is a good point, I added memoization which means the ids will have precedent over the names, I think that is fine for this (hopefully) rare case of passing both names and IDs.

One thing that could help this area is mention the `ssl_*_id` params are content credentials in the API docs. the name parameters and id parameters don't match:
```
 --ssl-ca-cert SSL_CA_CERT                                         Name of content credential containing CA certificate
 --ssl-ca-cert-id SSL_CA_CERT_ID                                   Idenifier of the SSL CA Cert
```
 Its not super clear right now (or giving any more clarity to the parameter name) that the SSL CA Cert id is an id of a content credential. There is also a typo :)

I will open a Katello PR for this!

Reply:
opened here: https://github.com/Katello/katello/pull/8546